connect/runtime/src/main/java/org/apache/kafka/connect/connector/policy/PrincipalConnectorClientConfigOverridePolicy.java

```diff
@@ -32,7 +32,9 @@
 /**
  * Allows all {@code sasl} configurations to be overridden via the connector configs by setting {@code connector.client.config.override.policy} to
  * {@code Principal}. This allows to set a principal per connector.
+ * @deprecated Use {@link AllowlistConnectorClientConfigOverridePolicy} instead.
  */
+@Deprecated(since = " 4.2", forRemoval = true)
```

Comment:
Can we add a warning log message when this class is instantiated to notify users about its deprecation and what steps they should take to prepare for its removal?